tests/test_schemas.py

```diff
@@ -0,0 +1,98 @@
+
+    Todo:
+        Return a v2.03 activity Schema.
+
+    """
+    return iati.default.activity_schema(version='2.02')
+
+
+def get_filepaths_in_folder(directory_name):
+    """Return the full filepaths of all xml files within the 'should-pass' folder.
+
+    Returns:
+        list of str: Filepaths within the 'should-pass' folder.
+
+    """
+    directory_name = directory_name + '**'
+    return [path for path in glob.glob(directory_name, recursive=True) if path.endswith(".xml")]
+
+
+def load_as_dataset(filepath):
+    """Load a filepath as an iati.Dataset object.
+
+    Args:
+        filepath (str): A relative path to an XML file.
+
+    Returns:
+        iati.Dataset: Representing the input XML file.
+
+    """
+    with open(filepath, 'r') as data:
+        xml_str = data.read()
+
+    return iati.Dataset(xml_str)
+
+
+@pytest.mark.parametrize('filepath', get_filepaths_in_folder('tests/activity-tests/should-pass/') + get_filepaths_in_folder('tests/should-pass/'))
+def test_pass_files(activity_schema, filepath):
+    """Check that all 'should-pass' test files are XML and pass Schema validation.
+
+    Todo:
+        Make this test work with a 2.03 activity Schema. Requires that pyIATI support version 2.03.
+
+    """
+    # Load the dataset
+    dataset = load_as_dataset(filepath)
+
+    # Attempt validation and assert valid
+    assert iati.validator.is_xml(dataset)
+    assert iati.validator.is_iati_xml(dataset, activity_schema)
+
+@pytest.mark.parametrize('filepath', get_filepaths_in_folder('tests/activity-tests/should-fail/'))
+def test_fail_files(activity_schema, filepath):
+    """Check that all 'should-fail' test files are XML but fail Schema validation.
+
+    Todo:
+        Make this test work with a 2.03 activity Schema. Requires that pyIATI support version 2.03.
+
+    """
+    # Load the dataset
+    dataset = load_as_dataset(filepath)
+
+    # Attempt validation and assert valid
+    assert iati.validator.is_xml(dataset)
+    assert not iati.validator.is_iati_xml(dataset, activity_schema)
+
+
+@pytest.mark.parametrize('filepath', get_filepaths_in_folder('tests/should-fail/'))
+def test_2_03_fail_files(activity_schema, filepath):
+    """Check that all 'should-fail' test files are XML but fail Schema validation for the expected reason.
+
+    Todo:
+        Make this test work with a 2.03 activity Schema. Requires that pyIATI support version 2.03.
+
+    """
+    # Load the dataset
+    dataset = load_as_dataset(filepath)
+
+    # Get the reason why this test should fail from the filename
+    # All filenames contain the pyIATI error name before the first underscore character
+    filename = os.path.split(filepath)[-1]
+    filename_no_extension = filename.split('.')[0]
+    failure_reason = filename_no_extension.split("_")[0]
+
+    # Attempt validation and assert valid
+    error_log = iati.validator.validate_is_iati_xml(dataset, activity_schema)
+
+    assert iati.validator.is_xml(dataset)
```

Comment:
I've a feeling that as-is, this should be called before `validate_is_iati_xml` (potentially a problem with pyIATI).

Reply:
This would break the AAA pattern though :-(

I'm of the understanding that pyIATI currently does not handle calls to `iati.validator.validate_is_iati_xml` gracefully if the input dataset is not well-formed XML. In my opinion it should though!

Reply:
Actually... this check shouldn't be needed since the Dataset shouldn't instantiate in the first place if the input string is not valid XML.

https://github.com/IATI/pyIATI/blob/f3c9b54dd3fb6d5e6fc48b3329e6b9a9682d1318/iati/data.py#L96-L105